### DISCUSSION TITLE
How do I use a hyphen in a prop name?

### DISCUSSION THREAD

**Discussion by unikitty37** [sveltejs/kit#10592] [Q&A]
I would like to pass the prop `data-testid` to a component. In Vue, I could define the prop as `dataTestid` and still use `data-testid` in the HTML tag, because Vue converts camel case to kebab case in these circumstances, but Svelte does not seem to support this?

I don't want the ugliness of using `data-testid` in HTML components and `dataTestid` in Svelte components. Is there a way to get Svelte to handle the hyphen correctly?

**Comment by teemingc** [comment]
I suggest moving this discussion to the svelte repository https://github.com/sveltejs/svelte

**Reply by unikitty37** [reply]
I was going to put it there, but there doesn't seem to be a "Discussions" section on that repo.

Looking at that repository, there _is_ an [issue dating back to 2019](https://github.com/sveltejs/svelte/issues/3852) requesting this be added, but the discussion seemed to get derailed by custom elements, aka web components — which I am not using.

The only solution that seems to work and not cause TypeScript complaints involves `$:` to pick it out of `$$props`, but that doesn't work with TypeScript as `data-testid` doesn't get added to the type signature — the code _runs_ but leaves big red complaints all over the editor window. (It also prevents me from configuring CI to not deploy if there are type errors…)

**Reply by teemingc** [reply]
Does https://github.com/roonie007/vite-plugin-svelte-kebab-props satisfy your requirements?

**Reply by unikitty37** [reply]
Alas, no — I tried it and got the same TypeScript complaints. It seems most people are using JavaScript and not noticing the issue :)
